sdk/security/keyvault/azsecrets/utils_test.go

```diff
@@ -37,7 +37,9 @@ var (
 )
 
 func TestMain(m *testing.M) {
-	vaultURL = strings.TrimSuffix(os.Getenv("AZURE_KEYVAULT_URL"), "/")
+	if recording.GetRecordMode() != recording.PlaybackMode {
+		vaultURL = strings.TrimSuffix(os.Getenv("AZURE_KEYVAULT_URL"), "/")
```

Comment:
Did this cause a problem in playback? Asking because at first glance it looks like it wouldn't make any difference, in which case I'd rather not have the conditional, if only to avoid making some future reader wonder about this 😄

Reply:
Yeah, it was causing an error on my end since I had the keyvault url set as an environment variable. Without the if statement, it'll pick up the real keyvault url instead of the fake one for the recordings. Guess we don't technically need it, but it's an easy issue to run into.

Reply:
I see, yes that would be a problem. You can tell I didn't let environment variables linger while I was working on this 🤭 

How about this:
```go
vaultURL = fakeVaultURL
if recording.GetRecordMode() != recording.PlaybackMode {
	if u, ok := os.LookupEnv("AZURE_KEYVAULT_URL"); ok {
		vaultURL = strings.TrimSuffix(u, "/")
	} else {
		panic("no value for AZURE_KEYVAULT_URL")
	}
}
```
(this reminds me of #18073)